server/server_test.go

```diff
@@ -1022,3 +1022,69 @@ func TestGetRandomIP(t *testing.T) {
 		}
 	}
 }
+
+type myConn struct {
+	sync.Mutex
+	net.Conn
+}
+
+func (mc *myConn) Write(b []byte) (int, error) {
+	mc.Lock()
```

Comment:
Don't need lock.. While implementing, I maintained some state in the struct, but not anymore, so will remove the lock.